lib/rubocop/cop/lint/raise_exception.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for `raise` or `fail` statements which is
+      # raise `Exception` class.
+      #
+      # @example
+      #   # bad
+      #   raise Exception, 'Error message here'
+      #
+      #   # good
+      #   raise StandardError, 'Error message here'
+      class RaiseException < Cop
+        MSG = 'Avoid `%<method>s` the `Exception` class. ' \
+              'Perhaps you meant to `%<method>s` `StandardError`?'
+
+        def_node_matcher :exception?, <<~PATTERN
+          (send nil? ${:raise :fail} (const _ :Exception) ... )
+        PATTERN
+
+        def_node_matcher :exception_new_with_message?, <<~PATTERN
+          (send nil? ${:raise :fail}
+            (send (const _ :Exception) :new ... ))
+        PATTERN
+
+        def on_send(node)
+          offense = exception?(node) || exception_new_with_message?(node)
+
+          add_offense(node, message: format(MSG, method: offense)) if offense
```

Comment:
You could combine the matchers and do:

```ruby
raise_exception?(node) do |method|
  add_offense(node, message: format(MSG, method: method))
end
```

Reply:
And how to combine them? I tried, but without success. You meant to write universal matcher or to combine two int one? 

or you meant  ? :) 

```
def raise_exception(node)
  exception?(node) || exception_new_with_message?(node)
end
```